### PR TITLE
Estiaan/termination entity and crud

### PR DESCRIPTION
**Pull request by madwizzard**
Screenshot 2024-06-05 133850

Screenshot 2024-06-05 151013

Unit Testing Report

Created a Save Endpoint which either Saves a new termination and updates employee Or upodates termination based on wether a termination already exists.

### CODE REVIEW

File: HRIS.Models/Enums/TerminationOption.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿namespace HRIS.Models.Enums;
+
+public enum TerminationOption
+{
```
Comment (tbddorothy) [inline HRIS.Models/Enums/TerminationOption.cs:4]: Why does this enum have no values?

File: HRIS.Services/Services/TerminationService.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        currentEmployee.TerminationDate = terminationDto.LastDayOfEmployment;
+        currentEmployee.Active = false;
+
+        EmployeeTypeDto employeeTypeDto = await _employeeTypeService
+           .GetEmployeeType(currentEmployee.EmployeeType!.Name);
+
+        await _db.Employee.Update(new Employee(currentEmployee, employeeTypeDto));
+        TerminationDto newTermination = await _db.Termination.Add(new Termination(terminationDto));
+
+
```
Comment (tbddorothy) [inline HRIS.Services/Services/TerminationService.cs:49]:
```suggestion
```

File: HRIS.Services/Services/TerminationService.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        if (!exists)
+        {
+            var exception = new Exception("termination not found");
+            throw _errorLoggingService.LogException(exception);
+        }
+
+        TerminationDto newTermination = await _db.Termination.FirstOrDefault(termination => termination.EmployeeId == employeeId);
+
+        return newTermination;
+
```
Comment (tbddorothy) [inline HRIS.Services/Services/TerminationService.cs:71]:
```suggestion
```

Comment (tbddorothy) [inline HRIS.Services/Services/TerminationService.cs:71]:
```suggestion
```

**Review by Juanro-Minne — DISMISSED** [review]

**Review by DewanAlberts — DISMISSED** [review]

**Review by Juanro-Minne — DISMISSED** [review]

File: HRIS.Services/Interfaces/ITerminationService.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace HRIS.Services.Interfaces;
+
+public interface ITerminationService
+{
+    /// <summary>
+    ///     Check if user exist
+    /// </summary>
+    /// <param name="email"></param>
+    /// <returns></returns>
```
Comment (MatthewSchoeman) [inline HRIS.Services/Interfaces/ITerminationService.cs:11]: The CheckTerminationExistst return of type bool but the returns in the comment above states nothing is returned
```suggestion
    /// <returns>Bool on whether the termination exists</returns>
```

File: HRIS.Services/Services/TerminationService.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        var exists = await CheckTerminationExist(employeeId);
+        if (!exists)
+        {
+            var exception = new Exception("termination not found");
+            throw _errorLoggingService.LogException(exception);
+        }
+
+        TerminationDto newTermination = await _db.Termination.FirstOrDefault(termination => termination.EmployeeId == employeeId);
+
+        return newTermination;
```
Comment (MatthewSchoeman) [inline HRIS.Services/Services/TerminationService.cs:61]: Is the newTermination variable necessary if the result can be returned without the need in storing the result?

**Review by tbddorothy — APPROVED** [review]

**Review by MatthewSchoeman — APPROVED** [review]

**Pull request merged**